src/message.c

```diff
@@ -680,33 +877,53 @@ int message_set_footer(MESSAGE_HANDLE message, annotations footer)
 {
 	int result;
 
-	if ((message == NULL) ||
-		(footer == NULL))
+	if (message == NULL)
 	{
-		LogError("Bad arguments: message = %p, footer = %p",
-			message, footer);
+		/* Codes_SRS_MESSAGE_01_077: [ If `message` is NULL, `message_set_footer` shall fail and return a non-zero value. ]*/
+		LogError("NULL message");
 		result = __FAILURE__;
 	}
 	else
 	{
-		AMQP_VALUE new_footer;
-
-		new_footer = annotations_clone(footer);
-		if (new_footer == NULL)
-		{
-			LogError("Cannot clone message footer");
-			result = __FAILURE__;
-		}
-		else
+		if (footer == NULL)
 		{
+			/* Codes_SRS_MESSAGE_01_151: [ If `footer` is NULL, the previously stored footer associated with `message` shall be freed. ]*/
 			if (message->footer != NULL)
 			{
 				annotations_destroy(message->footer);
```

Comment:
Done.

---
In reply to: [120404310](https://github.com/Azure/azure-uamqp-c/pull/128#discussion_r120404310) [](ancestors = 120404310)